common4j/src/main/com/microsoft/identity/common/java/controllers/CommandDispatcher.java

```diff
@@ -472,10 +474,14 @@ private static void logParameters(@NonNull String tag, @NonNull String correlati
                         + DiagnosticContext.INSTANCE.getRequestContext().toJsonString()
                         + ", with PublicApiId: " + publicApiId);
 
-        if (Logger.isAllowPii()) {
-            Logger.infoPII(TAG, ObjectMapper.serializeObjectToJsonString(parameters));
+        if (parameters instanceof BaseNativeAuthCommandParameters) {
```

Comment:
This check will introduce NativeAuth types to this class and all of its dependencies. All consumers of this library except NativeAuth have set up a proguard rule to exclude NativeAuth types. Due to the introduction of this type, all those consumers will now have to include NativeAuth byte code in their library. Their binary size will increase and they may not like that. If possible we should move this check to some other location where this problem will not occur.

Reply:
@SaurabhMSFT  good point, let me look into this.
@shahzaibj @fadidurah  what size checks can I run to show that Saurabh is right in this observation?

Reply:
@SammyO Can we update the check to be like the following:

```java
if (parameters instanceof ILoggable)
```

This should avoid anyone taking dependency on native auth

Reply:
Good point @SaurabhMSFT.

@fadidurah  @shahzaibj could you review the changes I made to address this? `logParameter` logic is now delegated to the CommandParameter classes.

Reply:
@shahzaibj sorry I missed your comment. I went in a different direction. Let me know if you're happy with this as well or whether I should revert to your suggestion.

Reply:
I think either approach is fine